### ISSUE TITLE
FB9724671: CGEventTap does not emit .mayBegin scroll events

### ISSUE THREAD

**Issue by chrenn** [feedback-assistant/reports#235]
- **Date:** 2021-10-26
- **Resolution:** Open
- **Area:** Core Graphics API
- **OS:** macOS 12.0.1
- **Type:** Incorrect/Unexpected Behavior
- **Keywords:** 

## Description

— Regression from macOS 11 to 12 —

In macOS Monterey, both CGEventTaps and NSEventMonitors are missing events with .type == .scrollWheel and .phase == .mayBegin. These events used to be emitted by Magic Trackpads on a tap & hold gesture with two fingers and were either followed by .phase == .began or .phase == .cancelled.

I also observed a decreased amount of events emitted during scrolling and pinching, which might be related but is pretty hard to document.

See the attached file for code and output. I used NSEvent.addGlobalMonitorForEvents() since it does not require accessibility privileges, but the same can be observed for CGEvent.tapCreate().

My app Swish (highlyopinionated.co/swish) makes heavy use of these scroll events and users have already complained after updating to Monterey. I really hope this is only a bug and not deactivated on purpose.

## Files

```swift
NSEvent.addGlobalMonitorForEvents(matching: .any) { event in
    if event.type == .scrollWheel {
        print(event)
    }
}

// TAP, HOLD, SCROLL & RELEASE

// Big Sur:
//NSEvent: type=ScrollWheel loc=(1298.94,173.355) time=710.7 flags=0 win=0x0 winNum=128 ctxt=0x0 deltaX=0.000000 deltaY=0.000000 count:0 phase=MayBegin momentumPhase=None
//NSEvent: type=ScrollWheel loc=(1298.94,173.355) time=711.9 flags=0 win=0x0 winNum=128 ctxt=0x0 deltaX=0.000000 deltaY=-1.000000 count:0 phase=Began momentumPhase=None
//NSEvent: type=ScrollWheel loc=(1298.94,173.355) time=711.9 flags=0 win=0x0 winNum=128 ctxt=0x0 deltaX=-1.000000 deltaY=-1.000000 count:0 phase=Changed momentumPhase=None
//...
//NSEvent: type=ScrollWheel loc=(1298.94,173.355) time=712.1 flags=0 win=0x0 winNum=128 ctxt=0x0 deltaX=0.000000 deltaY=-1.000000 count:0 phase=Changed momentumPhase=None
//NSEvent: type=ScrollWheel loc=(1298.94,173.355) time=712.2 flags=0 win=0x0 winNum=128 ctxt=0x0 deltaX=0.000000 deltaY=0.000000 count:0 phase=Ended momentumPhase=None

// Monterey:
//NSEvent: type=ScrollWheel loc=(1173.03,188.055) time=16894.2 flags=0 win=0x0 winNum=77 ctxt=0x0 deltaX=0.000000 deltaY=-1.000000 count:0 phase=Began momentumPhase=None
//NSEvent: type=ScrollWheel loc=(1173.03,188.055) time=16894.2 flags=0 win=0x0 winNum=77 ctxt=0x0 deltaX=0.000000 deltaY=-1.000000 count:0 phase=Changed momentumPhase=None
//...
//NSEvent: type=ScrollWheel loc=(1173.03,188.055) time=16894.5 flags=0 win=0x0 winNum=77 ctxt=0x0 deltaX=0.000000 deltaY=-1.000000 count:0 phase=Changed momentumPhase=None
//NSEvent: type=ScrollWheel loc=(1173.03,188.055) time=16894.8 flags=0 win=0x0 winNum=77 ctxt=0x0 deltaX=0.000000 deltaY=0.000000 count:0 phase=Ended momentumPhase=None


// TAP, HOLD & RELEASE

// Big Sur:
//NSEvent: type=ScrollWheel loc=(1232.66,147.719) time=749.7 flags=0 win=0x0 winNum=128 ctxt=0x0 deltaX=0.000000 deltaY=0.000000 count:0 phase=MayBegin momentumPhase=None
//NSEvent: type=ScrollWheel loc=(1232.66,147.719) time=750.4 flags=0 win=0x0 winNum=128 ctxt=0x0 deltaX=0.000000 deltaY=0.000000 count:0 phase=Cancelled momentumPhase=None

// Monterey:
// [no output]


```

**Comment by steliosaa** [comment]
Is this fixed in the new beta? 
(I think currently 12.1)

**Comment by steliosaa** [comment]
Any updates on this issue? Fixed on the new Monterey 12.1 beta or not yet?

**Comment by Arden144** [comment]
This issue appears to be fixed in Monterey 12.1 Beta 3
